### PR TITLE
Option to disable fake exp drops

### PR DESCRIPTION
**Pull request by robinwithes**
Just quickly drafted this up for a friend, tested it with him and there seem to be no issues with it.
Closes https://github.com/runelite/runelite/issues/7775

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java [hunk 1]
```diff
@@ -87,4 +87,14 @@ default int fakeXpDropDelay()
 		return 0;
 	}
 
+	@ConfigItem(
+		keyName = "disableFakeXpDrops",
+		name = "Disable fake xp drops",
+		description = "Enable to disable post 200m fake exp drops.",
```
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]:
```java
description = "Disables post-200m fake exp drops",
```

Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]: Don't ironmen also get this when attacking other players in the wilderness? Or did that not get implemented?

Comment (devLotto) [reply runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]: Also, you changed the wrong description

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropPlugin.java [hunk 2]
```diff
@@ -92,6 +92,16 @@ public void onWidgetHiddenChanged(WidgetHiddenChanged event)
 			return;
 		}
 
+		if (config.disableFakeXpDrops())
+		{
+			if (widget.getText().contains("img=11")) //fake xp drops always contain this
+			{
+				widget.getParent().setHidden(true);
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropPlugin.java]: Probably only this line is necessary?

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java (hunk 1 above)
Comment (Alexsuperfly) [reply runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]: Ironmen do receive these when attacking other players

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java [hunk 3]
```diff
@@ -79,12 +79,22 @@ default Color getMagePrayerColor()
 	@ConfigItem(
 		keyName = "fakeXpDropDelay",
 		name = "Fake Xp Drop delay",
-		description = "Configures how many ticks should pass between fake XP drops, 0 to disable",
+		description = "Disables post-200m fake exp drops",
```
Comment (TheStonedTurtle) [inline runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]: Why did you change this config?

Comment (robinwithes) [reply runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]: sleep deprevation

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java [hunk 4]
```diff
@@ -87,4 +87,14 @@ default int fakeXpDropDelay()
 		return 0;
 	}
 
+	@ConfigItem(
+		keyName = "disableFakeXpDrops",
+		name = "Disable fake xp drops",
+		description = "Disables post-200m fake exp drops",
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]: Mentioned this before in the other comment thread, this also applies to ironman XP drops gained from attacking other players in the wilderness.

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java [hunk 5]
```diff
@@ -87,4 +87,14 @@ default int fakeXpDropDelay()
 		return 0;
 	}
 
+	@ConfigItem(
+		keyName = "disableFakeXpDrops",
+		name = "Disable fake xp drops",
+		description = "Disables fake exp drops",
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]: I feel the extra explanation would be more useful to include than not.

```suggestion
		description = "Disables fake exp drops (such as post-200m or ironman pvp exp)",
```

**Comment by Adam-** [comment]
This seems fine if the history was squashed.

**Comment by robinwithes** [comment]
> 
> 
> This seems fine if the history was squashed.

Sorry for the late reply, i think it's sorted now.

**Comment by robinwithes** [comment]
https://twitter.com/JagexWolf/status/1100337130335887360
